File: packages/java/typescript-generator/src/test/java/com/vaadin/hilla/parser/plugins/transfertypes/jsonnode/JsonNodeEndpoint.java
package com.vaadin.hilla.parser.plugins.transfertypes.jsonnode;

import tools.jackson.databind.JsonNode;
import tools.jackson.databind.node.ArrayNode;
import tools.jackson.databind.node.ObjectNode;

@Endpoint
public class JsonNodeEndpoint {
    public JsonNode jsonNode(JsonNode node) {
        return node;
    }

    public ObjectNode objectNode(ObjectNode node) {
        return node;
    }

    public ArrayNode arrayNode(ArrayNode node) {
        return node;
    }
}

File: packages/java/typescript-generator/src/test/resources/com/vaadin/hilla/parser/plugins/transfertypes/jsonnode/snapshots/JsonNodeEndpoint.ts
import type { EndpointRequestInit } from '@vaadin/hilla-frontend';
import client from './connect-client.default.js';

export async function arrayNode(
  node: Array<unknown> | undefined,
  init?: EndpointRequestInit,
): Promise<Array<unknown> | undefined> {
  return client.call('JsonNodeEndpoint', 'arrayNode', { node }, init);
}

export async function jsonNode(node: unknown, init?: EndpointRequestInit): Promise<unknown> {
  return client.call('JsonNodeEndpoint', 'jsonNode', { node }, init);
}

export async function objectNode(node: unknown, init?: EndpointRequestInit): Promise<unknown> {
  return client.call('JsonNodeEndpoint', 'objectNode', { node }, init);
}
